meta/Decays.m

```diff
@@ -0,0 +1,2412 @@
+        0, 0, 1}, {0, 0, 1, 0, 0, 2}, {0, 0, 0, 1, 2, 0}}, "T8",
+      {{0, 0, 0, 1, 0, 0}, {0, 0, 0, 0, 1, 0}, {0, 0, 0, 1, 0, 0}, {1, 0, 1,
+        0, 0, 1}, {0, 1, 0, 0, 0, 2}, {0, 0, 0, 1, 2, 0}}, "T9",
+      {{0, 0, 0, 1, 0, 0}, {0, 0, 0, 0, 1, 0}, {0, 0, 0, 0, 1, 0}, {1, 0, 0,
+        0, 0, 2}, {0, 1, 1, 0, 0, 1}, {0, 0, 0, 2, 1, 0}}, "T10",
+      _, Print["Error: Cannot map topology to FeynArts name"]; Quit[1]
+    ];
+
+WrapCodeInLoop[indices_, code_] :=
+   (
+      "\n// loops over vertices' indices\n" <>
+      StringJoin@@(MapIndexed[
+      Nest[
+         TextFormatting`IndentText,
+         "for(const auto& index" <> ToString@#1 <> ": index_range<vertex" <> ToString@#1 <> ">()) {\n" <>
+            If[First@#2===Length[indices], "\n" <> TextFormatting`IndentText[code], ""],
+         First@#2 -1
+      ]&,
+      indices
+   ]) <>
+     StringJoin@@ (Nest[
+         TextFormatting`IndentText,
+         "}\n",
+         #-1
+         ]& /@ Reverse@Range@Length[indices]));
+
+WrapCodeInLoopOverInternalVertices[decay_, topology_, diagram_] :=
+   Module[{(*vertices, *)indices, cppVertices,
+      mass = {}, translation, fieldAssociation,
+      externalEdges,
+     (*verticesInFieldTypes, *)matchExternalFieldIndicesCode, matchInternalFieldIndicesCode = "", functionBody = "",
+     verticesInFieldTypesForFACp, verticesForFACp, colorFac = Unique["colorFac"], symmetryFac = Unique["symmetryFac"]
+   },
+
+      translation = GenericTranslationForInsertion[topology, diagram];
+      If[Length[translation]===7 && translation[[-3]] === {},
+        Return[{{}, ""}]
+      ];
+
+      (* the vertices positions FACp are not ordered according to numbers in element 2 & 3 of translation *)
+      (* vertex in terms of field types (S,V,F,...) and indices 1, 2 *)
+      (*      verticesInFieldTypes =*)
+      (*          Select[translation[[-2]], ListQ] /. Susyno`LieGroups`conj[f_] :> -f /. Field[n_] :> (Field[n]/.translation[[4]])[n];*)
+      verticesInFieldTypesForFACp =
+         List @@@ (DeleteDuplicates[translation[[-3]] /. Index[Generic, n_Integer] :> n /. Decays`Private`FACp[x__][__] :> FACp[x]]);
+
+      (* {Field[1] -> concrete field, ...} *)
+      fieldAssociation =
+          GetFieldsAssociations[
+            InsertionsOnEdgesForDiagram[topology, diagram],
+            translation[[3]],
+            translation[[4]],
+            diagram,
+             verticesInFieldTypesForFACp
+          ];
+
+
+      (* vertices in an orientation as required by Cp *)
+(*      vertices = verticesInFieldTypes /. (fieldAssociation /. ((#1 -> #2@@#1)& @@@ translation[[4]])) /. - e_ :> AntiField[e];*)
+      verticesForFACp = verticesInFieldTypesForFACp /. (fieldAssociation /. ((#1 -> #2@@#1)& @@@ translation[[4]])) /. - e_ :> AntiField[e];
+
+      (* set of unique indices used in names of vertices and indices *)
+      indices = Table[Unique["Id"], {Length@verticesForFACp}];
```

Comment:
For readability I'd propose to not use "random" numbers here. I'd propose to use the index scheme `index_1`, `index_2`, ... The same for the color and symmetry factors.

Reply:
Done in 5b7410ad67d240cda1b09034db587ba9c1cf19ef.

Reply:
thanks!